utils/vara-ui/src/components/checkbox/checkbox.module.scss

```diff
@@ -1,36 +1,198 @@
 .label {
   display: flex;
   align-items: center;
+  gap: 8px;
   cursor: pointer;
+  user-select: none;
 
-  &.disabled {
-    pointer-events: none;
-    opacity: 0.3;
+  &:has(.checkbox) {
+    position: relative;
+
+    &::after {
+      content: '';
+      position: absolute;
+      top: 3px;
+      left: 3px;
+      width: var(--checkbox-size);
+      height: var(--checkbox-size);
+      border-color: #00ffc466;
+      background-color: #000000;
+      mask-image: url("data:image/svg+xml,%3Csvg width='18' height='18' viewBox='0 0 18 18' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M2 9.00008L7 14.0001L16 5.00008L14.59 3.58008L7 11.1701L3.41 7.59008L2 9.00008Z' fill='black'/%3E%3C/svg%3E%0A");
+      opacity: 0;
+      transition-duration: 250ms;
+      transition-property: background-color, opacity;
+      transition-timing-function: ease;
+    }
+
+    &:has(:checked) {
+      &::after {
+        opacity: 1;
+      }
+
+      &:has(:disabled) {
+        &::after {
+          background-color: rgba(208, 213, 221, 1);
+        }
+      }
+    }
+  }
+
+  //Show disabled state even if an error displayed
+  &[aria-invalid=true]:not(.disabled) {
+    .checkbox {
+      border-color: rgba(255, 51, 51, 0.8);
+
+      &:checked {
+        background-color: rgba(255, 51, 51, 0.8);
+      }
+    }
```

Comment:
If we're targeting checkbox there, and not label, better to nest it in a `.checkbox` class itself, otherwise it's quite hard to read and follow the logic.